libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Conway/Instances.hs

```diff
@@ -1339,3 +1444,440 @@ onSized ::
+  propagateSpecFun ToDelta (NilCtx HOLE) (TypeSpec (NumSpecInterval l h) cant) =
+    ( TypeSpec
+        (NumSpecInterval (fromIntegral <$> l) (fromIntegral <$> h))
+        (map deltaToCoin cant)
+    )
+
+  mapTypeSpec ToDelta (NumSpecInterval l h) = typeSpec (NumSpecInterval (fromIntegral <$> l) (fromIntegral <$> h))
+
+deltaToCoin :: DeltaCoin -> Coin
+deltaToCoin (DeltaCoin i) = Coin i
+
+-- ===========================================================================
+
+-- \| EraPP era means we can go back and forth between PParamSubset and (PParams era)
+--   This allow us to use PParamsSubset as the (SimpleRep (PParams era))
+--   Much easier to constrain PParamSubset than (PParams era) with all the THKD stuff.
+class
+  ( Era era
+  , Eq (PParamsHKD Identity era)
+  , Show (PParamsHKD Identity era)
+  , Eq (PParamsHKD StrictMaybe era)
+  , Show (PParamsHKD StrictMaybe era)
+  ) =>
+  EraPP era
+  where
+  subsetToPP :: PParamSubset -> PParams era
+  ppToSubset :: PParams era -> PParamSubset
+  updateToPPU :: PPUpdate -> PParamsUpdate era
+  ppuToUpdate :: PParamsUpdate era -> PPUpdate
+
+instance EraPP Shelley where
+  subsetToPP = liftShelley
+  ppToSubset x = dropAtMost4 x $ dropShelley x
+  updateToPPU x = (uLiftProtVer x . uLiftShelley) x
+  ppuToUpdate x = uDropProtVer x $ uDropShelley x
+
+instance EraPP Allegra where
+  subsetToPP = liftShelley
+  ppToSubset x = dropAtMost4 x $ dropShelley x
+  updateToPPU x = (uLiftProtVer x . uLiftShelley) x
+  ppuToUpdate x = uDropProtVer x $ uDropShelley x
+
+instance EraPP Mary where
+  subsetToPP x = liftShelley x
+  ppToSubset x = dropAtMost4 x $ dropShelley x
+  updateToPPU x = (uLiftProtVer x . uLiftShelley) x
+  ppuToUpdate x = uDropProtVer x $ uDropShelley x
+
+instance EraPP Alonzo where
+  subsetToPP x = (liftAlonzo x . liftShelley) x
+  ppToSubset x = dropAlonzo x $ dropAtMost6 x $ dropShelley x
+  updateToPPU x = (uLiftAlonzo x . uLiftProtVer x . uLiftShelley) x
+  ppuToUpdate x = uDropAlonzo x $ uDropProtVer x $ uDropShelley x
+
+instance EraPP Babbage where
+  subsetToPP x = (liftBabbage x . liftAlonzo x . liftShelley) x
+  ppToSubset x = dropBabbage x $ dropAlonzo x $ dropShelley x
+  updateToPPU x = (uLiftBabbage x . uLiftAlonzo x . uLiftProtVer x . uLiftShelley) x
+  ppuToUpdate x = uDropBabbage x $ uDropAlonzo x $ uDropProtVer x $ uDropShelley x
+
+instance EraPP Conway where
+  subsetToPP x = (liftConway x . liftBabbage x . liftAlonzo x . liftShelley) x
+  ppToSubset x = dropConway x $ dropBabbage x $ dropAlonzo x $ dropShelley x
+  updateToPPU x = (uLiftConway x . uLiftBabbage x . uLiftAlonzo x . uLiftShelley) x
+  ppuToUpdate x = uDropConway x $ uDropBabbage x $ uDropAlonzo x $ uDropShelley x
+
+-- ===============================================================
+
+-- | SimpleRep instance for PParams
+instance EraPP era => HasSimpleRep (PParams era) where
+  type SimpleRep (PParams era) = PParamSubset
+  toSimpleRep = ppToSubset
+  fromSimpleRep = subsetToPP
```

Comment:
It might be worth while to add the selectors like
```haskell
minFeeA_ :: (IsConwayUniv fn, EraPP era) => Term fn (PParams era) -> Term fn Coin
minFeeA_ = sel @0
```